### PR TITLE
strip pod unused fields

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

Keep all that are used:  Labels, Container Ports, Status.Conditions,  Status.PodIPs

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 1]
```diff
@@ -1405,3 +1407,62 @@ func (c *Controller) AppendServiceHandler(f func(*model.Service, model.Event)) {
...
+func stripPodUnusedFields(obj any) (any, error) {
+	t, ok := obj.(metav1.ObjectMetaAccessor)
+	if !ok {
+		// shouldn't happen
+		return obj, nil
+	}
+	// ManagedFields is large and we never use it
+	t.GetObjectMeta().SetManagedFields(nil)
+	// Annotation is never used
+	t.GetObjectMeta().SetAnnotations(nil)
```
Comment (ramaraochavali) [inline pilot/pkg/serviceregistry/kube/controller/controller.go:1422]: Are we sure we will never use pod annotations in future also?

**Review by ramaraochavali — COMMENTED** [review]
LGTM with one question. Curious what is the reduction in memory?

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 2]
```diff
@@ -1405,3 +1407,62 @@ func (c *Controller) AppendServiceHandler(f func(*model.Service, model.Event)) {
...
+	// OwnerReference is never used
+	t.GetObjectMeta().SetOwnerReferences(nil)
```
Comment (howardjohn) [inline pilot/pkg/serviceregistry/kube/controller/controller.go:1424]: https://github.com/istio/istio/blob/b524d9836880a88c7aaa91956dfca24c82c29039/pkg/kube/util.go#L218

**Review by howardjohn — COMMENTED** [review]
Were you able to get any profiles to show real world improvement?

File: pilot/pkg/serviceregistry/kube/controller/controller.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/controller.go:1422]: At least not needed for now, not sure about future

File: pilot/pkg/serviceregistry/kube/controller/controller.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/controller.go:1424]: good catch,  missed it

**Comment by hzxuzhonghu** [comment]
I have tested 200 pods with https://gist.github.com/hzxuzhonghu/c930288e3d44fac7617dc0d85805f890

Below is the comparison

```
 ps -aux 
root      626246  0.1  0.1 1485468 25616 pts/0   Sl+  09:53   0:00 ./main --kubeconfig=/root/.kube/config
root      643790  0.4  0.2 1485720 37160 pts/2   Sl+  10:04   0:00 ./main-no-strip --kubeconfig=/root/.kube/config

```

**Review by ramaraochavali — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Pull request merged**